# Video Games Sales:

# Business Understanding¶
Here you need to pose three questions of your interest related to the dataset:
- What are the top 20 platforms in terms of the number of games?
- Distribution of games produced over the year of production.
- Top ten publishers in terms of number of games?
- What is the Global Sales of Video games each year?
- What is the top Video Game in the Global Sales?
- What is the top Video Game in the North Amireca Sales?
- What is the top Video Game in the Euorpe Sales?
- What is the top Video Game in the Japanese Sales?
- What is the top Video Game in the Other Sales?
- What is the total Global Sales for each ganre?

## Import the Libraris and Read the data

In [1]:
import numpy as np
import pandas as pd
import missingno as mn
from collections import Counter

#Visualization
import matplotlib.pyplot as plt
import matplotlib.style as style
from matplotlib.colors import ListedColormap
from matplotlib import cm
import seaborn as sns
%matplotlib inline

#Plotly Libraris
import plotly.express as px
import plotly.graph_objects as go
import plotly.figure_factory as ff
from plotly.colors import n_colors
from plotly.subplots import make_subplots
from plotly.offline import init_notebook_mode, iplot
from plotly import tools
from IPython.display import display, HTML

df = pd.read_csv('vgsales.csv')


Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

### Data Understanding
- show the head , tail, describe, info

In [134]:
df.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [2]:
df.tail()

Rank                                              Name Platform  \
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year     Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
16593  2002.0  Platform       Kemco      0.01      0.00       0.0   
16594  2003.0   Shooter  Infogrames      0.01      0.00       0.0   
16595  2008.0    Racing  Activision      0.00      0.00       0.0   
16596  2010.0    Puzzle    7G//AMES      0.00      0.01       0.0   
16597  2003.0  Platform     Wanadoo      0.01      0.00       0.0   

       Other_Sales  Global_Sales  
16593          0.0          0.01  
16594          0.0          0.01  
16595          0.0          0.01  
16596          0.0          0.01  
16597          0.0          0.01

In [135]:
df.describe()

Rank          Year      NA_Sales      EU_Sales      JP_Sales  \
count  16291.000000  16291.000000  16291.000000  16291.000000  16291.000000   
mean    8290.190228   2006.405561      0.265647      0.147731      0.078833   
std     4792.654450      5.832412      0.822432      0.509303      0.311879   
min        1.000000   1980.000000      0.000000      0.000000      0.000000   
25%     4132.500000   2003.000000      0.000000      0.000000      0.000000   
50%     8292.000000   2007.000000      0.080000      0.020000      0.000000   
75%    12439.500000   2010.000000      0.240000      0.110000      0.040000   
max    16600.000000   2020.000000     41.490000     29.020000     10.220000   

        Other_Sales  Global_Sales  
count  16291.000000  16291.000000  
mean       0.048426      0.540910  
std        0.190083      1.567345  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.040000      0.480000  
max       10.570000     82.740000

In [136]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16291 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16291 non-null  int64  
 1   Name          16291 non-null  object 
 2   Platform      16291 non-null  object 
 3   Year          16291 non-null  float64
 4   Genre         16291 non-null  object 
 5   Publisher     16291 non-null  object 
 6   NA_Sales      16291 non-null  float64
 7   EU_Sales      16291 non-null  float64
 8   JP_Sales      16291 non-null  float64
 9   Other_Sales   16291 non-null  float64
 10  Global_Sales  16291 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.5+ MB


In [174]:
games = df['Name'].unique()
publisher = df['Publisher'].unique()
platforms = df['Platform'].unique()
genres = df['Genre'].unique()

In [175]:

fig = go.Figure()
fig.add_trace(go.Indicator(
    mode = "number",
    value = len(df['Name'].unique()),
    title = {'text': "Games",'font': {'color': 'gold','size':20}},
    number={'font':{'color': 'gold','size':50}},
    domain = {'row': 0, 'column': 0}
))
fig.add_trace(go.Indicator(
    mode = "number",
    value = len(publisher),
    title = {'text': "Publishers",'font': {'color': 'green','size':20}},
    number={'font':{'color': 'green','size':50}},
    domain = {'row': 0, 'column': 1}
))

fig.add_trace(go.Indicator(
    mode = "number",
    value = len(platforms),
    title = {'text': "Platforms",'font': {'color': 'darkviolet','size':20}},
    number={'font':{'color': 'darkviolet','size':50}},
    domain = {'row': 1, 'column': 0}
))

fig.add_trace(go.Indicator(
    mode = "number",
    value = len(genres),
    title = {'text': "Genres",'font': {'color': 'deepskyblue','size':20}},
    number={'font':{'color': 'deepskyblue','size':50}},
    domain = {'row': 1, 'column': 1}
))

fig.update_layout(
    grid = {'rows': 2, 'columns': 2, 'pattern': "independent"})
fig.show()


### Prepare Data
Clean tha data
- missing value
- duplcated value

In [117]:
df.isna().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [132]:
df= df.dropna()

In [133]:
df.isna().sum()

Rank            0
Name            0
Platform        0
Year            0
Genre           0
Publisher       0
NA_Sales        0
EU_Sales        0
JP_Sales        0
Other_Sales     0
Global_Sales    0
dtype: int64

In [137]:
df.duplicated().sum()

0

### What are the 20 ten platforms in terms of the number of games?

In [161]:
plat =  df.groupby('Platform')['Name'].count().reset_index()
plat = plat.sort_values('Name',ascending=False).reset_index()
plat.drop("index",axis = 1,inplace=True)

In [162]:
fig = go.Figure()

# Add Traces


fig.add_trace(
    go.Bar(x=plat['Platform'][:20],
           y=plat['Name'][:20],
           name="Top 10",
          marker={'color': pubwisegame['Name'][:10],'colorscale': 'sunset'}))
# Set title
fig.update_layout(
    title_text="Platform Wise Video Game Count",
    xaxis_title="Platform",
    yaxis_title="Video Game Count",
    xaxis_domain=[0.05, 1.0]
)


Finding:
We Found the highest platform in the Video Game Count is Ds then PS2

### Distribution of games produced over the year of production.

In [167]:
years =  df.groupby('Year')['Name'].count().reset_index()
years = years.sort_values('Name',ascending=False).reset_index()
years.drop("index",axis = 1,inplace=True)

In [168]:
years.head ()

Year  Name
0  2009.0  1431
1  2008.0  1428
2  2010.0  1257
3  2007.0  1201
4  2011.0  1136

In [169]:
fig = go.Figure()

# Add Traces


fig.add_trace(
    go.Bar(x=years['Year'],
           y=years['Name'],
           name="Year",
          marker={'color': years['Name'],'colorscale': 'sunset'}))
# Set title
fig.update_layout(
    title_text="Years Wise Video Game Count",
    xaxis_title="Year",
    yaxis_title="Video Game Count",
    xaxis_domain=[0.05, 1.0]
)


Finding: 
We Found that the top year in Video Game Count is 2009(1431 Gmase) then 2008 (1428 games) 

### Top 20 publishers in terms of number of games?

In [170]:
publish = df.groupby('Publisher')['Name'].count().reset_index()
publish = publish.sort_values('Name',ascending=False).reset_index()
publish.drop("index",axis = 1,inplace=True)

In [172]:
fig = go.Figure()

# Add Traces


fig.add_trace(
    go.Bar(x=publish['Publisher'][:20],
           y=publish['Name'][:20],
           name="Publisher",
          marker={'color': publish['Name'][:20],'colorscale': 'sunset'}))
# Set title
fig.update_layout(
    title_text="Publisher Wise Video Game Count",
    xaxis_title="Publisher",
    yaxis_title="Video Game Count",
    xaxis_domain=[0.05, 1.0]
)


Finding:
We found that the top publishers in terms of total games is Electronic Arts (1339 games) then Activision (966 games)

### What the count of Global Sales of Video games each year?

In [176]:
yearwisesale =  df.groupby('Year')['Global_Sales'].sum().reset_index()

In [177]:
fig = go.Figure(go.Bar(x=yearwisesale['Year'],y=yearwisesale['Global_Sales'],
                       marker={'color': yearwisesale['Global_Sales'],'colorscale': 'sunset'}))
fig.update_layout(title_text='Video Game Global Sales by Release Year in ',xaxis_title="Year",yaxis_title="Sum of Sales")
fig.show()

Finding:
We found that the top year in terms of total public sales of video games is 2008 (678.9 B) then 2009 (667.3 B)

### What the top Video Game in the Global Sales?

In [179]:
df[df['Global_Sales']==df['Global_Sales'].max()]

Rank        Name Platform    Year   Genre Publisher  NA_Sales  EU_Sales  \
0     1  Wii Sports      Wii  2006.0  Sports  Nintendo     41.49     29.02   

   JP_Sales  Other_Sales  Global_Sales  
0      3.77         8.46         82.74

Finding:
the top of video games in the Global Sales is Wii Sports (82.74)

### What the top Video Game in the North Amireca Sales?

In [180]:
df[df['NA_Sales']==df['NA_Sales'].max()]

Rank        Name Platform    Year   Genre Publisher  NA_Sales  EU_Sales  \
0     1  Wii Sports      Wii  2006.0  Sports  Nintendo     41.49     29.02   

   JP_Sales  Other_Sales  Global_Sales  
0      3.77         8.46         82.74

Finding: the top of video games in the North Amireca Sales is Wii Sports (41.49)

### What the top Video Game in the Euorpe Sales?

In [181]:
df[df['EU_Sales']==df['EU_Sales'].max()]

Rank        Name Platform    Year   Genre Publisher  NA_Sales  EU_Sales  \
0     1  Wii Sports      Wii  2006.0  Sports  Nintendo     41.49     29.02   

   JP_Sales  Other_Sales  Global_Sales  
0      3.77         8.46         82.74

Finding: the top of video games in the Euorpe Sales is Wii Sports (29.02)

### What the top Video Game in the Japanese Sales?

In [12]:
df[df['JP_Sales']==df['JP_Sales'].max()]

Rank                      Name Platform    Year         Genre Publisher  \
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
4     11.27      8.89     10.22          1.0         31.37

Finding: the top of video games in the Japanese Sales is Pokemon Red/Pokemon Blue (10.22)

### What the top Video Game in the Other Sales?

In [16]:
df[df['Other_Sales']==df['Other_Sales'].max()]

Rank                           Name Platform    Year   Genre  \
17    18  Grand Theft Auto: San Andreas      PS2  2004.0  Action   

               Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
17  Take-Two Interactive      9.43       0.4      0.41        10.57   

    Global_Sales  
17         20.81

Finding: the top of video games in the Other Sales is Grand Theft Auto: San Andreas (10.57)

### What is the total Global Sales for each ganre?

In [185]:
ganre =  df.groupby('Genre')['Global_Sales'].sum().reset_index()
ganre = ganre.sort_values('Global_Sales',ascending=False).reset_index()
ganre.drop("index",axis = 1,inplace=True)

In [186]:
fig = go.Figure(go.Bar(x=ganre['Genre'],y=ganre['Global_Sales'],
                       marker={'color': ganre['Global_Sales'],'colorscale': 'sunset'}))
fig.update_layout(title_text='Video Game Global Sales by Genre ',
                  xaxis_title="Genre",
                  yaxis_title="Sum of Sales")
fig.show()

Finding: The top of Ganres in the Global Sales is Action (1722.84) then Sports (1309.24)

### Conclusion:

In this dataset we have 11.3 K Games , 576 Publisher , 31 Platform and 12 Ganre,the highest platform in the Video Game Count is Ds then PS2 and the top year in Video Game Count is 2009(1431 Gmase) then 2008 (1428 games)and the top publishers in terms of total games is Electronic Arts (1339 games) then Activision (966 games)and the top year in terms of total public sales of video games is 2008 (678.9 B) then 2009 (667.3 B) and the top of video games in the Global Sales, North Amireca Sales and Euorpe Sales is Wii Sports (82.74)and the top of video games in the Japanese Sales is Pokemon Red/Pokemon Blue (10.22) and the top of video games in the Other Sales is Grand Theft Auto: San Andreas (10.57) and The top of Ganres in the Global Sales is Action (1722.84) then Sports (1309.24).

In [187]:
#export the dataframe to csv file:
df.to_csv('dfvideo_games.csv')